src/types/index.ts

```diff
@@ -251,58 +216,52 @@ export enum ClaimType {
   InvestorUniquenessV2 = 'InvestorUniquenessV2',
 }
 
+export type AccreditedClaim = { type: ClaimType.Accredited; scope: Scope };
+
+export type AffiliateClaim = { type: ClaimType.Affiliate; scope: Scope };
+
+export type BuyLockupClaim = { type: ClaimType.BuyLockup; scope: Scope };
+
+export type SellLockupClaim = { type: ClaimType.SellLockup; scope: Scope };
+
 export type CddClaim = { type: ClaimType.CustomerDueDiligence; id: string };
 
+export type KnowYourCustomerClaim = { type: ClaimType.KnowYourCustomer; scope: Scope };
+
+export type JurisdictionClaim = { type: ClaimType.Jurisdiction; code: CountryCode; scope: Scope };
+
+export type ExemptClaim = { type: ClaimType.Exempted; scope: Scope };
+
+export type BlockedClaim = { type: ClaimType.Blocked; scope: Scope };
+
 export type InvestorUniquenessClaim = {
   type: ClaimType.InvestorUniqueness;
   scope: Scope;
   cddId: string;
   scopeId: string;
 };
 
+export type NoDataClaim = { type: ClaimType.NoData };
+
 export type InvestorUniquenessV2Claim = {
```

Comment:
All of these should be interfaces. Even though in current typescript versions there aren't many differences between type and interface, there still exist some, and we're using interface wherever possible. Better be consistent

Reply:
Interfaces lack the operations like `&` and `|`, which we use for `ScopedClaim`, `UnscopedClaim` and `Claim`.

We can express the base types as interfaces, but the mentioned ones need to be types.

I think its more consistent to leave them all as types, unless there is a way to define the unions as interfaces, which I don't think is the case. I checked a few blogs and all the examples use type for the union part.

Reply:
Even though it's more consistent within the claim types to use `type`, it's more consistent with the code base at large to use `interface`

The rule of thumb we're using is "use interface unless it's not possible" (for example, when using unions, as you said)

Reply:
I'll change the ones I can to interface, but leave the compound ones as types.

Personally I think having them all types feels more consistent, but the rule of thumb is reasonable enough. Especially when there isn't much difference between them.

Reply:
There used to be bigger differences. For example, types didn't show up by name in compilation errors sometimes, among other things. It's more of a legacy thing + the current typescript manual recommends it so in a certain way it could be considered language convention 😅